### DISCUSSION TITLE
【Bug?】Wrong result when using Subquery in SELECT clause

### DISCUSSION THREAD

**Discussion by hiroyuki0415** [MarkMpn/Sql4Cds#315] [Q&A]
```sql
SELECT r.crf3c_rank0_3id,
       r.crf3c_integer_column,
       r.owningbusinessunit,
       (SELECT count(*)
        FROM   crf3c_rank0_3 AS sub
        WHERE  sub.crf3c_integer_column <= r.crf3c_integer_column) AS cnt
FROM   crf3c_rank0_3 AS r;
```

The result of this SQL is below. this result is correct:

<HTML><head></head>
<BODY>
<TABLE><THEAD><TH>&nbsp;</TH><TH>crf3c_rank0_3id</TH><TH>crf3c_integer_column</TH><TH>owningbusinessunit</TH><TH>cnt</TH></THEAD><TR><TD ALIGN="center">&nbsp;</TD><TD>2f0384fb-5b14-ee11-9cbe-002248627e12</TD><TD>2</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>2</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>63aecf0d-5c14-ee11-9cbe-002248627e12</TD><TD>5</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>5</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>13a93ef8-5b14-ee11-9cbe-002248627ef1</TD><TD>1</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>1</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>b51e0406-5c14-ee11-9cbe-002248627ef1</TD><TD>3</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>3</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>08a15f0d-5c14-ee11-9cbe-002248627ef1</TD><TD>4</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>4</TD></TR></TABLE>

</BODY>
</HTML>

But when I add one condition to the subquery, it returns wrong result:

```sql
SELECT r.crf3c_rank0_3id,
       r.crf3c_integer_column,
       r.owningbusinessunit,
       (SELECT count(*)
        FROM   crf3c_rank0_3 AS sub
        WHERE  sub.owningbusinessunit = r.owningbusinessunit --add other condition
               AND sub.crf3c_integer_column <= r.crf3c_integer_column) AS cnt
FROM   crf3c_rank0_3 AS r;

```

wrong result: 

<HTML><head></head>
<BODY>
<TABLE><THEAD><TH>&nbsp;</TH><TH>crf3c_rank0_3id</TH><TH>crf3c_integer_column</TH><TH>owningbusinessunit</TH><TH>cnt</TH></THEAD><TR><TD ALIGN="center">&nbsp;</TD><TD>2f0384fb-5b14-ee11-9cbe-002248627e12</TD><TD>2</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>2</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>63aecf0d-5c14-ee11-9cbe-002248627e12</TD><TD>5</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>2</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>13a93ef8-5b14-ee11-9cbe-002248627ef1</TD><TD>1</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>2</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>b51e0406-5c14-ee11-9cbe-002248627ef1</TD><TD>3</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>2</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>08a15f0d-5c14-ee11-9cbe-002248627ef1</TD><TD>4</TD><TD>866c34a9-2021-ec11-b6e6-002248622ce3</TD><TD>2</TD></TR></TABLE>

</BODY>
</HTML>

Thanks in advance for your support.

**Comment by MarkMpn** [comment]
Thanks, yes this appears to be a bug which is triggered when you combine an `=` correlated condition in the subquery with another condition. I've opened issue #316 to track this.
